pandas/core/tools/datetimes.py

```diff
@@ -316,8 +317,20 @@ def _convert_listlike_datetimes(
     elif unit is not None:
         if format is not None:
             raise ValueError("cannot specify both format and unit")
-        arg = getattr(arg, "values", arg)
-        result, tz_parsed = tslib.array_with_unit_to_datetime(arg, unit, errors=errors)
+        arg = getattr(arg, "_values", arg)
+        # GH 30050 pass an ndarray to tslib.array_with_unit_to_datetime
+        # because it expects an ndarray argument
+        if isinstance(arg, IntegerArray):
+            # Explicitly pass NaT mask to array_with_unit_to_datetime
+            mask_na = arg.isna()
+            arg = getattr(arg, "_ndarray_values", arg)
+            result, tz_parsed = tslib.array_with_unit_to_datetime(
```

Comment:
move this outside the if/else and add ``mask=None`` for the non-IntegerArray case, then you can just call this once